lib/msprime.c

```diff
@@ -5839,8 +5841,8 @@ beta_integrand(double x, void *p)
     for (l = 0; l <= m; l++)  {
         ret += exp(r[l] - r_max);
     }
-    ret = 1 - exp(r_max + log(ret));
-    ret *= exp((-1 - alpha) * log_x + (alpha - 1) * log_1_minus_x);
+    exponent = (-1 - alpha) * log_x + (alpha - 1) * log_1_minus_x;
```

Comment:
Can we not add `log(4) - gsl_sf_lnbeta(2 - alpha, alpha)` to `exponent` and delete line 5846? That is probably more stable for `alpha` close to 2 since it avoids computing a near-singular beta function.

Reply:
Yep, done.